### PR TITLE
netsuite QA fixes

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

File: src/app/core/models/common/import-settings.model.ts [hunk 1]
```diff
@@ -89,12 +89,11 @@ export class ImportSettingsModel {
...
     });
 
+
     accountingAppFields.forEach((accountingAppField) => {
-      if (expenseFieldFormArray.length < 3) {
-        const fieldData = unmappedFieldMap.get(accountingAppField.attribute_type);
-        if (fieldData) {
-          expenseFieldFormArray.push(this.createFormGroup(fieldData));
-        }
+      const fieldData = unmappedFieldMap.get(accountingAppField.attribute_type);
```
Comment (ashwin1111) [inline src/app/core/models/common/import-settings.model.ts:94]: Intacct behaviour would change, there we display 3 fields at max.

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.html [hunk 2]
```diff
@@ -9,7 +9,7 @@
                 [contentText]="brandingContent.contentText"
                 [redirectLink]="redirectLink"
                 [showSyncButton]="isOnboarding"
-                (refreshDimensions)="refreshDimensions()">
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.html:12]: has to be fixed for qbo export setting as well then

File: src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.html [hunk 3]
```diff
@@ -17,8 +17,8 @@
                 <div class="tw-rounded-lg tw-border-border-tertiary tw-border">
                     <app-configuration-toggle-field
                         [form]="importSettingForm"
-                        [label]="brandingContent.importCategoriesLabel"
-                        [subLabel]="brandingContent.importCategoriesSubLabel"
+                        [label]="(brandingConfig.brandId === 'co' ? (getCategoryLabel() | sentenceCase) : getCategoryLabel()) + brandingContent.importCategoriesLabel"
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.html:20]: let's do all conversions in ts pls

File: src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.ts [hunk 4]
```diff
@@ -178,6 +178,21 @@ export class NetsuiteImportSettingsComponent implements OnInit {
     this.helperService.refreshNetsuiteDimensions().subscribe();
   }
 
+  getCategoryLabel(): string {
+    if (this.importSettings.configuration.import_netsuite_employee) {
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.ts:182]: check employee field mapping from configurations

File: src/app/core/models/sage300/sage300-configuration/sage300-import-settings.model.ts [hunk 5]
```diff
@@ -51,7 +51,7 @@ export class Sage300ImportSettingModel extends ImportSettingsModel {
     }
 
     static mapAPIResponseToFormGroup(importSettings: Sage300ImportSettingGet | null, sage300Fields: IntegrationField[]): FormGroup {
-        const expenseFieldsArray = importSettings?.mapping_settings ? this.constructFormArray(importSettings.mapping_settings, sage300Fields) : [] ;
+        const expenseFieldsArray = importSettings?.mapping_settings ? this.constructFormArray(importSettings.mapping_settings, sage300Fields, false) : [] ;
```
Comment (ashwin1111) [inline src/app/core/models/sage300/sage300-configuration/sage300-import-settings.model.ts:54]: we should send false from Intacct as well

**Review by ashwin1111 — APPROVED** [review]

File: src/app/core/models/sage300/sage300-configuration/sage300-import-settings.model.ts (hunk 5 above)
Comment (DhaaraniCIT) [reply src/app/core/models/sage300/sage300-configuration/sage300-import-settings.model.ts:54]: @ashwin1111 for intacct both fyle and C1 theme are not using the function in common import settings model, it has its own `constructFormArray()`

Screenshot 2024-05-07 at 8 34 36 AM

Screenshot 2024-05-07 at 8 35 07 AM

**Pull request merged**